src/scores/continuous/correlation/correlation_impl.py

```diff
@@ -0,0 +1,54 @@
+"""
+This module contains functions to calculate the correlation.
+"""
+
+from typing import Optional
+
+import xarray as xr
+
+import scores.utils
+from scores.typing import FlexibleDimensionTypes
+
+
+def pearsonr(
+    fcst: xr.DataArray,
+    obs: xr.DataArray,
+    *,  # Force keywords arguments to be keyword-only
+    reduce_dims: Optional[FlexibleDimensionTypes] = None,
+    preserve_dims: Optional[FlexibleDimensionTypes] = None,
+) -> xr.DataArray:
+    """
+    Calculates the Pearson's correlation coefficient between two xarray DataArrays
```

Comment:
If correlation can't take weights, should we mention that?

Reply:
Added a note